e2e.Dockerfile

```diff
@@ -1,20 +1,32 @@
 ARG IMG_TAG=latest
 
 # Compile the gaiad binary
-FROM golang:1.21-alpine AS gaiad-builder
+FROM golang:1.21-alpine3.18 AS gaiad-builder
 WORKDIR /src/app/
+ENV PACKAGES="curl make git libc-dev bash file gcc linux-headers eudev-dev python3"
+RUN apk add --no-cache $PACKAGES
+
+# See https://github.com/CosmWasm/wasmvm/releases
+ARG WASMVM_VERSION=v1.5.0
+ADD https://github.com/CosmWasm/wasmvm/releases/download/${WASMVM_VERSION}/libwasmvm_muslc.aarch64.a /lib/libwasmvm_muslc.aarch64.a
+ADD https://github.com/CosmWasm/wasmvm/releases/download/${WASMVM_VERSION}/libwasmvm_muslc.x86_64.a /lib/libwasmvm_muslc.x86_64.a
+RUN sha256sum /lib/libwasmvm_muslc.aarch64.a | grep 2687afbdae1bc6c7c8b05ae20dfb8ffc7ddc5b4e056697d0f37853dfe294e913
+RUN sha256sum /lib/libwasmvm_muslc.x86_64.a | grep 465e3a088e96fd009a11bfd234c69fb8a0556967677e54511c084f815cf9ce63
+RUN cp "/lib/libwasmvm_muslc.$(uname -m).a" /lib/libwasmvm_muslc.a
+
 COPY go.mod go.sum* ./
 RUN go mod download
 COPY . .
-ENV PACKAGES curl make git libc-dev bash gcc linux-headers eudev-dev python3
-RUN apk add --no-cache $PACKAGES
-RUN CGO_ENABLED=0 make install
+RUN LEDGER_ENABLED=false LINK_STATICALLY=true BUILD_TAGS=muslc make build
```

Comment:
Had issues with this in e2e.

CGO needs to be enabled (don't know why).
I've also had to add `RUN apk add --no-cache build-base
` in the runner container to make CGO available.

It seems that we cannot run the tests with CGO enabled.

I'm not sure what are the implications of this. While checking other projects, they all seem to run with CGO enabled (or at least they don't eplicitly disable it using `CGO_ENABLED=0`.

Reply:
> This repo contains both Rust and Go code. The Rust code is compiled into a library (shared .dll/.dylib/.so or static .a) to be linked via cgo and wrapped with a pleasant Go API

CGO is a requirement for CW yes https://github.com/CosmWasm/wasmvm